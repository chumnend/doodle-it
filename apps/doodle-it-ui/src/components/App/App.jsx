import { Routes, Route } from 'react-router';

import Landing from '../Landing';
import Designer from '../Designer';
import Home from '../Home';
import Register from '../Register';
import Login from '../Login';
import Logout from '../Logout';
import NotFound from '../NotFound';

import * as PATH from '../../helpers/constants';

const App = () => {
  return (
    <Routes>
      <Route path={PATH.HOME} element={< Home />} />
      <Route path={PATH.DESIGNER} element={<Designer />} />
      <Route path={PATH.REGISTER} element={<Register />} />
      <Route path={PATH.LOGIN} element={<Login />} />
      <Route path={PATH.LOGOUT} element={<Logout />} />
      <Route index element={<Landing />} />
      <Route path="*" element={<NotFound />} />
    </Routes>
  );
};

export default App
