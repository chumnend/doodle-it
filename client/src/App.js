import React, { useState, useEffect, useCallback } from 'react';
import { Switch, Route } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import Nav from './components/Nav';
import ProtectedRoute from './components/ProtectedRoute';
import Designer from './containers/Designer';
import Home from './containers/Home';
import Register from './containers/Register';
import Landing from './containers/Landing';
import Login from './containers/Login';
import Logout from './containers/Logout';
import NotFound from './containers/NotFound';
import * as ROUTES from './constants/routes';
import * as actions from './store/actions';

const App = () => {
  const [loading, setLoading] = useState(true);
  const auth = useSelector((state) => state.auth);
  const dispatch = useDispatch();

  const validateUser = useCallback(
    () => dispatch(actions.authRequestValidate()),
    [dispatch],
  );

  useEffect(() => {
    validateUser();
    setLoading(false);
  }, [validateUser]);

  return (
    <>
      <Nav isLoggedIn={!!auth.token} />
      {!loading && (
        <Switch>
          <ProtectedRoute
            exact
            path={ROUTES.HOME}
            condition={!!auth.token}
            redirect={ROUTES.LOGIN}
            component={Home}
          />
          <ProtectedRoute
            exact
            path={ROUTES.DESIGNER}
            condition={!!auth.token}
            redirect={ROUTES.LOGIN}
            component={Designer}
          />
          <ProtectedRoute
            exact
            path={ROUTES.LANDING}
            condition={!auth.token}
            redirect={ROUTES.HOME}
            component={Landing}
          />
          <ProtectedRoute
            exact
            path={ROUTES.REGISTER}
            condition={!auth.token}
            redirect={ROUTES.HOME}
            component={Register}
          />
          <ProtectedRoute
            exact
            path={ROUTES.LOGIN}
            condition={!auth.token}
            redirect={ROUTES.HOME}
            component={Login}
          />
          <Route exact path={ROUTES.LOGOUT} component={Logout} />
          <Route component={NotFound} />
        </Switch>
      )}
    </>
  );
};

export default App;
