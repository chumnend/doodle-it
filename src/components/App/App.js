import React from 'react';
import "./App.css";

function App() {
    return (
        <div className="App">
            <h1>MERN React Template</h1>
            <h4>By Nicholas Chumney</h4>
        </div>
    );
}

export default App;
