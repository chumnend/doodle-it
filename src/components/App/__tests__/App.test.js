import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

describe("APP COMPONENT", function () {
    test("expects to render a component", function () {
        const { container, getByText } = render(<App />);
        expect(getByText('MERN React Template')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });  
});
