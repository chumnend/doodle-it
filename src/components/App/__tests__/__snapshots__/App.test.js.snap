// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP COMPONENT expects to render a component 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      MERN React Template
    </h1>
    <h4>
      By Nicholas Chumney
    </h4>
  </div>
</div>
`;
